Доделать приложение с урока, дописать методы обновления и удаления по…

gemansipator opened the pull request:
…льзователя + провести

gemansipator committed:
Доделать приложение с урока, дописать методы обновления и удаления пользователя + провести

RinatBeybutov commented on src/main/java/site/javadev/springsecuritydemo1508home/controllers/AuthController.java:
```diff
@@ -40,62 +41,51 @@ public AuthController(PeopleService peopleService, JWTUtil jwtUtil, ModelMapper
 
+    @GetMapping("/user/{id}")
+    public Optional<PersonDTO> getUserById(@PathVariable Long id) {
+        return peopleService.findById(id).map(peopleService::convertPersonToDTO);
```
конвертацию в дто нужно сделать в сервисе, а почему не возвращается ResponseEntity?

RinatBeybutov commented on src/main/java/site/javadev/springsecuritydemo1508home/model/Person.java:
```diff
@@ -29,6 +29,18 @@ public class Person {
+    @Column(name = "email")
+    private String email;
+
+    public String getEmail() {
```
почему не использовать @Getter и @Setter?

RinatBeybutov commented on src/main/java/site/javadev/springsecuritydemo1508home/model/Person.java:
```diff
@@ -29,6 +29,18 @@ public class Person {
+    }
+
+
     public Person() {
```
@NoArgsConstructor?

gemansipator committed:
конвертация в сервисе и прописаны аннотации

gemansipator committed:
✔ Добавлены подробные комментарии к методам
✔ Объяснены проверки при регистрации
✔ Объяснен процесс аутентификации и генерации JWT
✔ Добавлены комментарии к обработке ошибок

RinatBeybutov commented on src/main/java/site/javadev/springsecuritydemo1508home/controllers/AuthController.java:
```diff
@@ -28,64 +24,115 @@ public class AuthController {
 
     @Autowired
-    public AuthController(PeopleService peopleService, JWTUtil jwtUtil, ModelMapper mapper, PersonValidator personValidator, AuthenticationManager authenticationManager) {
+    public AuthController(PeopleService peopleService, JWTUtil jwtUtil, AuthenticationManager authenticationManager) {
```
можно заменить на @RequiredArgsConstrictor если JWTUtil является бином и убрать этот конструктор

RinatBeybutov commented on src/main/java/site/javadev/springsecuritydemo1508home/controllers/AuthController.java:
```diff
@@ -28,64 +24,115 @@ public class AuthController {
+        }
+
+        // Преобразуем DTO в сущность Person и сохраняем в базе
         Person person = peopleService.convertDTOToPerson(personDTO);
```
лучше в сервис передавать дто, а там преобразовывать и сохранять в базу

gemansipator committed:
Объяснение изменений:
AuthController:

Мы заменили явное объявление конструктора на аннотацию @RequiredArgsConstructor, которая сгенерирует конструктор автоматически для всех полей с final.
Логика преобразования DTO и сохранения объекта была передана в сервисный слой в метод registerPerson. Это сделано для того, чтобы контроллер был сосредоточен только на обработке HTTP-запросов.
PeopleService:

Мы добавили новый метод registerPerson, который принимает DTO, преобразует его в сущность и сохраняет её. Это позволяет контроллеру просто передавать DTO в сервис и не заниматься преобразованием и сохранением.
